protostar/commands/cairo1_commands/fetch_from_scarb.py

```diff
@@ -0,0 +1,136 @@
+        )
+    except subprocess.CalledProcessError as ex:
+        raise ScarbMetadataFetchException(str(ex)) from ex
+
+    if result.returncode != 0:
+        raise ScarbMetadataFetchException(
+            "Metadata fetch returned a non-zero exit code."
+        )
+
+    # only the last line of the output contains metadata
+    result = result.stdout.strip().split(b"\n")[-1]
+
+    try:
+        return json.loads(result)
+    except json.JSONDecodeError as ex:
+        raise ScarbMetadataFetchException("Failed to decode the metadata json.") from ex
+
+
+def maybe_fetch_linked_libraries(project_root_path: Path) -> Optional[List[Path]]:
+    if not has_scarb_toml(project_root_path):
+        logging.info(
+            "Scarb.toml not found, using only packages provided by the argument."
+        )
+        return None
+
+    logging.info("Scarb.toml found, fetching Scarb packages.")
+
+    scarb_toml_path = project_root_path / "Scarb.toml"
+    metadata = read_scarb_metadata(scarb_toml_path)
+
+    try:
+        # assuming we have only one entry in the workspace section
+        current_package_name = metadata["workspace"]["members"][0]
+
+        matching_compilation_units = list(
+            filter(
+                lambda compilation_unit: compilation_unit["package"]
+                == current_package_name,
+                metadata["compilation_units"],
+            )
+        )
+
+        matching_contract_units = list(
+            filter(
+                lambda compilation_unit: compilation_unit["target"]["kind"]
+                == "starknet-contract",
+                matching_compilation_units,
+            )
+        )
+        matching_lib_units = list(
+            filter(
+                lambda compilation_unit: compilation_unit["target"]["kind"] == "lib",
+                matching_compilation_units,
+            )
+        )
+
+        #  lib is the default target
+        unit_with_dependencies = (
+            matching_contract_units[0]
+            if matching_contract_units
+            else matching_lib_units[0]
+        )
+
+        if len(matching_contract_units) > 1:
+            logging.warning(
+                "Scarb found multiple starknet-contract targets - using the target named %s.",
+                {unit_with_dependencies["target"]["name"]},
+            )
+
+        valuable_dependencies = filter(
+            lambda dependency: dependency["name"] != "core",
+            unit_with_dependencies["components_data"],
+        )
+
+        # using path.parent because component["source_path"] points to lib.cairo
+        paths: List[Path] = list(
+            map(
+                lambda component: find_directory_with_cairo_project_toml(
+                    Path(component["source_path"])
+                ),
+                valuable_dependencies,
+            )
+        )
+
+    except (IndexError, KeyError) as ex:
+        raise ScarbMetadataFetchException("Error parsing metadata:\n" + str(ex)) from ex
```

Comment:
catching general exceptions in a broad scope may lead to unexpected errors

Reply:
Agree, but I don't see how can we make it more specific in this case.